parcels/particlefile/particlefileaos.py

```diff
@@ -111,6 +113,13 @@ def export(self):
             if MPI.COMM_WORLD.Get_rank() > 0:
                 return  # export only on threat 0
 
+        # Create dictionary to translate datatypes and fill_values
+        self.fmt_map = {np.float32: 'f4', np.float64: 'f8',
+                        np.bool_: 'i1', np.int16: 'i2', np.int32: 'i4', np.int64: 'i8'}
+        self.fill_value_map = {np.float32: np.nan, np.float64: np.nan,
+                               np.bool_: np.iinfo(np.bool_).max, np.int16: np.iinfo(np.int16).max,
```

Comment:
There seems to be a bug in `np.bool_: np.iinfo(np.bool_).max`. Max value for a bool is perhaps a rather strange concept?
Anyways, when I changed locally to np.nan, it worked
```suggestion
                               np.bool_: np.nan, np.int16: np.iinfo(np.int16).max,
```

Reply:
Looks good - for my future reference, is there a way to run these tests locally so I can avoid pushing modifications with silly mistakes?

Reply:
Yes, you can run tests locally with the `pytest` from command line (e.g. `pytest tests/*.py`)

Reply:
@nvogtvincent yes, `np.nan` and `np.bool` makes conceptually no sense. `np.nan` and any non-float datatype is anyway non-sense, because integer datatypes can express any integer numbers - there are no integer-NaN's. There is something like '_overflow_', but that's basically just a binary clip after <x> bits. Hence, `nan` and `inf` are purely floating-point concepts. Numpy internally though is smart enough to map `np.nan` to `0` for integers.

I think it would be smarter to cleanly define the fill-value for bool: after all, it can only be `0` or `1`, and in this context setting it to `0` makes more sense than to `1`.

Reply:
@CKehl But this fill value is for the netcdf file, which can't handle boolean datatypes anyway? The closest netcdf has to a boolean (as far as I'm aware) is an 8-bit integer, in which case I think it would be most consistent to have the fill value set to `np.iinfo(np.int8).max` not `0` (which would surely be incorrect anyway since a `False` boolean is not the same thing as the absence of data)?

Reply:
yes, indeed @nvogtvincent - I am also not arguing about the `i1` choice ;-) So, 8-bit (signed) integer is fine.
The point with the fill value is that it is a (somehow) _valid_ value in the datatype you represent information. In the file, the boolean would be 8-bit integer, so `np.iinfo(np.int8).max` would be fine. But if you then read the file into an array again, then `np.iinfo(np.int8).max` will be an overflow and will result in auto-`1` (or auto-`True`) the value. I don't know how good that maps the intention. Perhaps @erikvansebille has some feelings about it.

But overall, the 'fill value for bool' question is really boiling down to viewpoint, because in the end you really only have the choice of the 2 values `True` and `False` - and deciding for one also automatically means deciding against the other. Personally, I'd rather choice for `0` than `1`, cause after all what's otherwise the purpose of expressing `1` (i.e. `True`) anyway ...

Reply:
Sorry @CKehl, I'm a bit confused what you mean by "read the file into an array again". If you load this 8-bit integer variable from the netcdf into python (or whatever), it should load it as a (masked) 8-bit integer array. You'd only get an overflow if you attempt to convert the unmasked array (raw data) to a boolean datatype. I completely agree that this consequence of converting fill values to `True` is not good and happy to see what @erikvansebille thinks about this but from my perspective, being able to distinguish between missing values and `False` values, albeit with the warning that some care must be taken when reading the file, is preferable to not being able to distinguish between between missing values and `False` values at all!

Reply:
Practical point response ;-):
- masked array conversion or clipping: depends on how the data are written. As we write it as contiguous array per attribute, you're right with the masking. This, though, it makes no difference. The only valid values for bools are `0 (False)` or `1(True)`.
- because of boolean being boolean, there is by definition no space to have a missing value skewing the data, so there will not be a way to distinguish the missing value either from a defined `True` (if decided to use `1` as missing value) or separating it from `False` (if deciding to use `0` as missing value). One of the two options is will be.
- on the warning: good point. Would you actually see it as 'important; to be informed about it for every simulation ?